### PR TITLE
Output

### PR DESCRIPTION
**Pull request by yramachers**
New primitive scorer to record global hit time at each collision (default [ns]), similar to location scorer. Took out unit in G4double for simple storage in ntuple.
Added event printout of entries in each HC at end of event for checking. 
The time scorer is not used anywhere just yet.

### CODE REVIEW

**Comment by drbenmorgan** [comment]
Hi @yramachers, as expected and noted above there are conflicts. They can be resolved either through the [GitHub "Resolve Conflicts" button](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/resolving-a-merge-conflict-on-github), or to do so manually:

```console
$ git checkout output
$ git fetch origin
$ git rebase origin/master
... git will now complain about conflicts...
```

GitHub help also [has an article on resolving things manually](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/resolving-a-merge-conflict-using-the-command-line). The only addendum to that is _not_ to follow step 8, but run:

```console
$ git rebase --continue
```

as the last step. As you have three commits, its possible there'll be additional conflicts, so just follow the above for each step.

I prefer rebasing to merging, but both work (it doesn't hurt to learn rebasing if you're not familiar with it though!).

File: src/WLGDPSLocation.cc [hunk 1]
```diff
@@ -5,14 +5,12 @@
 ////////////////////////////////////////////////////////////////////////////////
 // Description:
 //   This is a primitive scorer class for scoring energy deposit location
-//   at termination of the track.
 //
 ///////////////////////////////////////////////////////////////////////////////
 
 WLGDPSLocation::WLGDPSLocation(G4String name, G4int depth)
 : G4VPrimitiveScorer(name, depth)
```
Comment (drbenmorgan) [inline src/WLGDPSLocation.cc:12]:
```suggestion
: G4VPrimitiveScorer(std::move(name), depth)
```

avoids an extra copy

File: src/WLGDPSLocation.cc [hunk 2]
```diff
@@ -21,7 +19,6 @@ WLGDPSLocation::WLGDPSLocation(G4String name, G4int depth)
 WLGDPSLocation::WLGDPSLocation(G4String name, const G4String& unit, G4int depth)
 : G4VPrimitiveScorer(name, depth)
```
Comment (drbenmorgan) [inline src/WLGDPSLocation.cc:20]:
```suggestion
: G4VPrimitiveScorer(std::move(name), depth)
```

File: src/WLGDPSTime.cc [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+#include "G4UnitsTable.hh"
+
+////////////////////////////////////////////////////////////////////////////////
+// Description:
+//   This is a primitive scorer class for scoring global track time
+//
+///////////////////////////////////////////////////////////////////////////////
+
+WLGDPSTime::WLGDPSTime(G4String name, G4int depth)
+: G4VPrimitiveScorer(name, depth)
```
Comment (drbenmorgan) [inline src/WLGDPSTime.cc:12]:
```suggestion
: G4VPrimitiveScorer(std::move(name), depth)
```

File: src/WLGDPSTime.cc [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+, HCID(-1)
+, EvtMap(0)
+{
+    SetUnit("ns");
+}
+
+WLGDPSTime::WLGDPSTime(G4String name, const G4String& unit, G4int depth)
+: G4VPrimitiveScorer(name, depth)
```
Comment (drbenmorgan) [inline src/WLGDPSTime.cc:20]:
```suggestion
: G4VPrimitiveScorer(std::move(name), depth)
```

**Comment by yramachers** [comment]
The web button merge resolver was very easy to follow, much better than my editor would be. Anyway, all resolved. I'll make sure for the next branches to always run clang format as last thing before pushing. 
I think, I'll next try to write a scorer that records the parentID and the vertex location since I want to record where the track came from that did something interesting. I reckon with a particle filter set to consider only neutrons and the main ion 77Ge, we might get all the interesting information.

File: src/WLGDPSTime.cc [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+, HCID(-1)
+, EvtMap(0)
+{
+    SetUnit(unit);
+}
+
+WLGDPSTime::~WLGDPSTime() { ; }
```
Comment (drbenmorgan) [inline src/WLGDPSTime.cc:27]:
```suggestion
WLGDPSTime::~WLGDPSTime() = default;
```

Or declare as `default` in the header

File: src/WLGDPSTime.cc [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+G4bool WLGDPSTime::ProcessHits(G4Step* aStep, G4TouchableHistory*)
+{
+    if(aStep->GetPostStepPoint()->GetStepStatus() == fGeomBoundary)
+        return FALSE;  // all collisions
```
Comment (drbenmorgan) [inline src/WLGDPSTime.cc:32]:
```suggestion
    // better comment here - what does "all collisions" mean?
    if(aStep->GetPostStepPoint()->GetStepStatus() == fGeomBoundary)
    {
        return FALSE;
    }
```

Comment (yramachers) [reply src/WLGDPSTime.cc:32]: comes straight from the collision primitive scorer, it is the geometry boundary status. Anything not a step to a boundary is a collision.

**Comment by drbenmorgan** [comment]
Thanks @yramachers, will merge assuming CI passes!

**Pull request merged**